samplePDFDUNE/samplePDFDUNEBeamND.cpp

```diff
@@ -0,0 +1,467 @@
+      em_escale_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(em_escale_nd_pos);
+    }
+    else if (name == "EMEScaleSqrtND") {
+      em_escale_sqrt_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(em_escale_sqrt_nd_pos);
+    }
+    else if (name == "EMEScaleInvSqrtND") {
+      em_escale_invsqrt_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(em_escale_invsqrt_nd_pos);
+    }
+    else if (name == "HadResND") {
+      had_res_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(had_res_nd_pos);
+    }
+    else if (name == "MuResND") {
+      mu_res_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(mu_res_nd_pos);
+    }
+    else if (name == "NResND") {
+      n_res_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(n_res_nd_pos);
+    }
+    else if (name == "EMResND") {
+      em_res_nd_pos = *it;
+      NDDetectorSystPointers[func_it] = XsecCov->retPointer(em_res_nd_pos);
+    }
+    else { 
+      MACH3LOG_ERROR("Found a functional parameter which wasn't specified in the xml | samplePDFDUNEBeamND: {}",name);
+      throw MaCh3Exception(__FILE__, __LINE__);
+    }
+  }
+  
+  std::cout << "-------------------------------------------------------------------" <<std::endl;
+}
+
+void samplePDFDUNEBeamND::SetupSplines() {
+  ///@todo move all of the spline setup into core
+  if(XsecCov->GetNumParamsFromDetID(SampleDetID, kSpline) > 0){
+    MACH3LOG_INFO("Found {} splines for this sample so I will create a spline object", XsecCov->GetNumParamsFromDetID(SampleDetID, kSpline));
+    splinesDUNE* DUNESplines = new splinesDUNE(XsecCov);
+    splineFile = (splineFDBase*)DUNESplines;
+    InitialiseSplineObject();
+  }
+  else{
+    MACH3LOG_INFO("Found {} splines for this sample so I will not load or evaluate splines", XsecCov->GetNumParamsFromDetID(SampleDetID, kSpline));
+    splineFile = nullptr;
+  }
+
+  return;
+}
+
+void samplePDFDUNEBeamND::SetupWeightPointers() {
+  for (int i = 0; i < (int)dunendmcSamples.size(); ++i) {
+    for (int j = 0; j < dunendmcSamples[i].nEvents; ++j) {
+      MCSamples[i].ntotal_weight_pointers[j] = 6;
+      MCSamples[i].total_weight_pointers[j] = new const double*[MCSamples[i].ntotal_weight_pointers[j]];
+      MCSamples[i].total_weight_pointers[j][0] = &(dunendmcSamples[i].pot_s);
+      MCSamples[i].total_weight_pointers[j][1] = &(dunendmcSamples[i].norm_s);
+      MCSamples[i].total_weight_pointers[j][2] = MCSamples[i].osc_w_pointer[j];
+      MCSamples[i].total_weight_pointers[j][3] = &(dunendmcSamples[i].rw_berpaacvwgt[j]);
+      MCSamples[i].total_weight_pointers[j][4] = &(dunendmcSamples[i].flux_w[j]);
+      MCSamples[i].total_weight_pointers[j][5] = &(MCSamples[i].xsec_w[j]);
+    }
+  }
+}
+
+int samplePDFDUNEBeamND::setupExperimentMC(int iSample) {
+  const char *sampleFile = (mtupleprefix+mtuple_files[iSample]+mtuplesuffix).c_str();
```

Comment:
this is wrong, it should be 

```char const *sampleFile = mtuple_files[iSample].c_str()```

looking at how the config is parsed in base and FD/Atm DUNE samplePDFs

Reply:
Fixed - you are correct